### PR TITLE
Add compute tag for elastic strain

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

The strain won't be explicitly solved for anymore with the upcoming second-order DG operator, so this compute tag is used to observe it and dependent quantities such as the potential energy.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/PointwiseFunctions/Elasticity/Test_Strain.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  tnsr::I<DataVector, Dim> displacement{x.begin()->size()};
+  get<0>(displacement) = square(get<0>(x)) + 2. * get<1>(x);
+  get<1>(displacement) = square(get<1>(x)) + 3. * get<0>(x);
+  if constexpr (Dim == 3) {
+    get<2>(displacement) = square(get<2>(x)) + get<0>(x) + 4. * get<1>(x);
+  }
+  return displacement;
+}
+
+template <size_t Dim>
```
Comment (nikwit) [inline tests/Unit/PointwiseFunctions/Elasticity/Test_Strain.cpp:41]: a short comment that this is $\nabla_{(i} \xi_{j)}$ of polynomial_displacement might be helpful

**Review by nikwit — COMMENTED** [review]
looks good to me. Do I understand correctly that nabla here just refers to a purely spatial gradient and not a covariant derivative?

**Comment by nilsvu** [comment]
@nikwit It can actually be a covariant derivative, though we don't currently solve elasticity systems on a curved geometry. But for the XCTS solves we'll have a strain that's the symmetrized _covariant_ derivative of the shift vector: `B_{ij}=\nabla_{(i}\gamma_{j)k}\beta^{k}`

**Comment by nikwit** [comment]
@nilsleiffischer  Yes, I see. But the way it is calculated here with `partial_derivatives`, it would not be able to take into account non-zero Christoffel symbols, right?

**Comment by nilsvu** [comment]
That's right, to support a curved geometry we would have to add an overload here that takes a metric and christoffel symbols.

**Comment by nikwit** [comment]
ok, thanks :) One more thing: the affine map in the unit test here would be an edge case where the jacobian is constant over the domain (is independent of coordinates) if I understand that correctly. But I assume thats fine as we are not testing the calculation of the partial derivatives itself.

**Comment by nilsvu** [comment]
Yep that's how I see it, too

**Review by kidder — COMMENTED** [review]
I have no requested changes

**Comment by nilsvu** [comment]
Rebased and squashed in the one added comment. Thanks for your reviews @nikwit and @kidder!

**Review by kidder — DISMISSED** [review]

**Comment by nilsvu** [comment]
@kidder @nikwit rebased again and added some missing includes.

**Review by kidder — APPROVED** [review]

**Review by nikwit — APPROVED** [review]
